docs/book/command-params.md

```diff
@@ -1,62 +1,277 @@
+  `VALUE_NONE` or `VALUE_REQUIRED`).
+- `getQuestion()`: Each parameter type will define its own `Question` to return.
+
+Another trait, `Laminas\Cli\Input\StandardQuestionTrait`, composes the
+`InputParamTrait` and adds the method `createQuestion()`, which returns a
+`Symfony\Component\Console\Question\Question` instance with a prompt in the form
+of:
+
+```text
+<question>{description}</question> [<comment>{default}</comment>]:
+>
+```
+
+This allows an implementation to use the standard format question, and then add
+things such as [normalizers](https://symfony.com/doc/current/components/console/helpers/questionhelper.html#normalizing-the-answer),
+[validators](https://symfony.com/doc/current/components/console/helpers/questionhelper.html#validating-the-answer),
+or [autocompletion](https://symfony.com/doc/current/components/console/helpers/questionhelper.html#autocompletion).
+
+## Standard input parameter types
+
+We ship several standard input parameter types for use in your applications. All
+parameters require the parameter name as the initial argument, and additional
+arguments as specified below.
+
+### BoolParam
+
+`Laminas\Cli\Input\BoolParam` allows specifying a parameter with a boolean
+value. It emits a `Symfony\Component\Console\Question\ConfirmationQuestion`.
+
+```php
+namespace Laminas\Cli\Input;
+
+final class BoolParam implements InputParamInterface
+{
+    use InputParamTrait;
+
+    public function __construct(string $name);
+}
+```
+
+### ChoiceParam
+
+`Laminas\Cli\Input\ChoiceParam` allows specifying a set of choices from which
+the user may select a value, emitting a `Symfony\Component\Console\Question\ChoiceQuestion`.
+
+```php
+namespace Laminas\Cli\Input;
+
+final class ChoiceParam implements InputParamInterface
+{
+    use StandardQuestionTrait;
+
+    public function __construct(string $name, array $haystack);
+}
+```
+
+### IntParam
+
+`Laminas\Cli\Input\IntParam` allows specifying that a value must be an integer,
+and optionally be more than a minimum value and/or less than a maximum value.
+
+```php
+namespace Laminas\Cli\Input;
+
+final class IntParam implements InputParamInterface
+{
+    use StandardQuestionTrait;
+
+    public function __construct(string $name);
+
+    public function setMin(?int $min): self;
+
+    public function setMax(?int $max): self;
+}
+```
+
+### PathParam
+
+`Laminas\Cli\Input\PathParam` allows specifying that a value must be a path on
+the filesystem, optionally requiring that it exist or be one of either a
+directory or file.
+
+```php
+namespace Laminas\Cli\Input;
+
+final class PathParam implements InputParamInterface
+{
+    use StandardQuestionTrait;
+
+    public const TYPE_DIR  = 'dir';
+    public const TYPE_FILE = 'file';
+
+    public function __construct(string $name);
+
+    /**
+     * @param string $type One of the TYPE_* constants
+     */
+    public function setPathType(string $type): self;
+
+    public function pathMustExist(bool $flag): self;
```

Comment:
I am not sure about non-standard setter name. I think it is good to keep the convention, and have all setters with `set*`.